### PR TITLE
venv usage MERGE parsing form excel files + ramp function 

### PR DESCRIPTION
**Pull request by jorgeboucas**
This fork allows parsing form excel files for non yaml lovers and generation of ramps which are specially nice for short warm ups and cool downs.

### CODE REVIEW

**Comment by mkuthan** [comment]
Wow, thanks for contribution! A few days ago I migrated to plain python venv so now there is a conflict in the Pipfile file :( Would you mind to update the pull request and move dependencies to requirements.txt file?

File: garminworkouts/__main__.py [hunk 1]
```diff
@@ -14,7 +16,8 @@
 def command_import(args):
     workout_files = glob.glob(args.workout)
 
-    workout_configs = [configreader.read_config(workout_file) for workout_file in workout_files]
+    workout_configs = [configreader.read_config(workout_file, args.ftp) for workout_file in workout_files]
+    # sys.exit(0)
```
Comment (mkuthan) [inline garminworkouts/__main__.py:20]: It should be removed

File: garminworkouts/config/excelparser.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        end=df.loc[i,"end"]
+        duration=df.loc[i,"duration"]
+        if str(end) == str(np.nan):
+            return '  - {{ power: {power}, duration: "{duration}" }}\n'.format(power=round(start,2), duration=duration)
+        else:
+            if start == end:
+                return '  - {{ power: {power}, duration: "{duration}" }}\n'.format(power=round(start,2), duration=duration)
+            elif start != end:
+                seconds=int(duration.split(":")[-1])+int(duration.split(":")[-2])*60
+                n_blocks=seconds/seconds_block
```
Comment (mkuthan) [inline garminworkouts/config/excelparser.py:41]: Are you aware of maximum number of steps in the single Garmin workout? If I remember the limit is around 50.

File: garminworkouts/models/power.py [hunk 3]
```diff
@@ -32,4 +32,4 @@ def _has_percent(self):
 
     @staticmethod
     def _to_absolute(power, ftp):
-        return int(power) * ftp / 100
+        return float(power) * ftp / 100
```
Comment (mkuthan) [inline garminworkouts/models/power.py:35]: What is the reason of this change?

**Review by mkuthan — CHANGES_REQUESTED** [review]
Only minor issues or rather questions.

File: garminworkouts/config/excelparser.py (hunk 2 above)
Comment (jorgeboucas) [reply garminworkouts/config/excelparser.py]: read about that today; will take your word for it. 
solved in 5999552

File: garminworkouts/__main__.py (hunk 1 above)
Comment (jorgeboucas) [reply garminworkouts/__main__.py]: solved in 5999552

File: garminworkouts/models/power.py (hunk 3 above)
Comment (jorgeboucas) [reply garminworkouts/models/power.py]: not sure why it should be int :-) 
solved in 5999552

**Comment by jorgeboucas** [comment]
ready to merge

**Review by jorgeboucas — COMMENTED** [review]
all issues have been addressed

**Review by mkuthan — APPROVED** [review]
